src/runner.ts

```diff
@@ -100,3 +99,28 @@ export const createRunner = <T>({
 export const createRunner$ = <T>(authOpts?: AuthOpts) => (
   query: Query
 ) => defer(() => from(createRunner<T>(authOpts)(query)))
+
+export const createCsvRunner = ({
```

Comment:
Why are we going with a separate query runner for CSV? Can we not use the runner? Ideally we can use the same runner to any arbitrary data type. Otherwise in the future, we’ll end up having loads of other runner to add support for something like geo json and more

Reply:
@walleXD Their were type issues with trying to return data Json data and csv data in the same runner. It As long as we know what the data inside a geo json looks like we can just use the createRunner function

Reply:
Then let’s take a compositional approach. Let’s create a raw runner, which jus makes a response and returns the raw got response. Then we can create wrapper runners which take the raw runner’s response and parses the data. That way the code is dry as all the fetching logic is in the rawRunner and parsing logic for each type of data is in their associated runners

Reply:
@walleXD Wouldn't the end result be the same though... Either way we would have to create a parser for each new type of data set.

Reply:
@walleXD so the raw runner function would return a function that parses the raw data ?